File: frontend/src/types/rule.ts
export interface RuleSuggestionResponse {
    id: string;
    rule_type: string;
    pattern: string;
    replacement: string;
    confidence: number;
    reasoning: string;
    examples: string[];
    priority: number;
    timestamp: string;
    suggested_rule?: {
      rule_type: string;
      pattern: string;
      replacement: string;
    };
  }
  
  export interface RuleDecisionRequest {
    decision: 'approve' | 'reject' | 'modify';
    reasoning: string;
    modifications: Record<string, any>;
  }
  
  export interface RuleResponse {
    id: string;
    rule_type: string;
    pattern: string;
    replacement: string;
    confidence: number;
    active: boolean;
    created_at: string;
    created_by: string;
    performance_score?: number;
  }
  
  export interface RulePerformanceResponse {
    rule_id: string;
    applications_count: number;
    success_rate: number;
    average_confidence: number;
    improvement_impact: number;
    trend: string;
    last_30_days: any[];
  }

File: web/models/rule.py
from pydantic import BaseModel
from typing import List, Dict, Any, Optional
from datetime import datetime
from enum import Enum

class RuleType(str, Enum):
    COMPANY = "company"
    MEASUREMENT = "measurement" 
    MATERIAL = "material"
    FORMATTING = "formatting"

class RuleDecision(str, Enum):
    APPROVE = "approve"
    REJECT = "reject"
    MODIFY = "modify"

class RuleSuggestionResponse(BaseModel):
    id: str
    rule_type: str
    pattern: str
    replacement: str
    confidence: float
    reasoning: str
    examples: List[str]
    priority: int
    timestamp: datetime
    suggested_rule: Optional[Dict[str, str]] = None

class RuleDecisionRequest(BaseModel):
    decision: RuleDecision
    reasoning: str
    modifications: Dict[str, Any] = {}

class RuleResponse(BaseModel):
    id: str
    rule_type: str
    pattern: str
    replacement: str
    confidence: float
    active: bool
    created_at: datetime
    created_by: str
    performance_score: Optional[float] = None

class RuleRequest(BaseModel):
    rule_type: str
    pattern: str
    replacement: str
    description: Optional[str] = None
    priority: int = 5

class RulePerformanceResponse(BaseModel):
    rule_id: str
    applications_count: int
    success_rate: float
    average_confidence: float
    improvement_impact: float
    trend: str
    last_30_days: List[Dict[str, Any]]